crates/matrix-sdk-base/src/store/mod.rs

```diff
@@ -243,6 +241,25 @@ impl Store {
         Ok(())
     }
 
+    /// Similar to [`Store::set_session_meta_and_reload_state`] but takes values
+    /// from another existing [`Store`].
+    #[cfg(any(feature = "e2e-encryption", test))]
+    pub(crate) async fn set_session_meta_and_reload_state_from_other(
+        &self,
+        other: &Self,
+        room_info_notable_update_sender: &broadcast::Sender<RoomInfoNotableUpdate>,
+    ) -> Result<()> {
```

Comment:
Instead of introducing a new function, could the caller call the existing `set_session_meta` with the session_meta extracted from the other store, instead? I don't see the point of duplicating the function when we could compose function calls instead :thinking:

Reply:
Because I'm going to add a new parameter, as the commit description mentions. Instead of exposing the internal data and let the caller deriving/composing with them, I believe it's a better design to have a dedicated method for that. It's also unit testable and documentable.